Service controller

lena-aiu opened the pull request:
Hi John, please review the changes in service_controller.rb, where I tried to create an access controller. I'm not sure that it's 100% right, because for example used (admin? || intern?) instead of (current_user.role.include?("administrator") || current_user.role.include?("intern")) @jrmcgarvey

jrmcgarvey reviewed (commented):
This is basically correct, except for the one error identified below, but I think there is significant risk in merging this PR so soon before the demo, so I will not merge it right now.  Please fix the error so we can merge later.  I will also add that the code is kind of verbose -- one could reduce some of the logic and if/else statements.

lena-aiu commented on app/controllers/services_controller.rb:
```diff
@@ -53,13 +100,35 @@ def create
     else
       flash.now.alert = @service.errors.full_messages.to_sentence
-      render :edit
+      render :new
```
Okay, thank you

jrmcgarvey replied:
This is still render :new it looks like.  It should be render :edit

lena-aiu commented:
I just push changes that I made base on your comments @jrmcgarvey

lena-aiu commented:
@jrmcgarvey Please review the code. I fixed errors with the request tests for services, making changes in the access controller for services. Also fixed an error with the request test for categories, uncommenting the index method in categories_controller.rb.

jrmcgarvey commented on app/controllers/categories_controller.rb:
```diff
@@ -61,18 +61,18 @@ def create
     else
       flash.now.alert = @category.errors.full_messages.to_sentence
-      render :new  
+      render :new
```
This shoudl be render :edit

jrmcgarvey commented on spec/requests/services_spec.rb:
```diff
@@ -105,7 +105,7 @@
       expect { post services_path, params: {service: service_attributes}
     }.to_not change(Service, :count)
-      expect(response).to render_template(:edit)
+      expect(response).to render_template(:new)
```
this should be render_template(:edit)

jrmcgarvey reviewed (changes requested):
There are three places where we have :new where it should be :edit.

jrmcgarvey commented on app/controllers/services_controller.rb:
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
   def new
-    @service = Service.new
+    # @service = Service.new
+    if  nil? || volunteer?
```
The if nil? here is not correct.  I think what is meant is, if current_user.role.nil? .  If nil? just means, is the current instance of the services_controller nil, which is always false.  In any case, these three lines are unnecessary.  You should take them out and just start with if admin? || intern?

lena-aiu replied:
Okay, I will make changes

jrmcgarvey commented on app/controllers/services_controller.rb:
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
   # GET /services/1/edit
   def edit
-
+    if  nil? || volunteer?
```
Again, the if nil? is not correct, and these 3 lines ae not needed.

lena-aiu replied:
Okay, I will make changes

jrmcgarvey commented on app/controllers/services_controller.rb:
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
   # POST /services
   # POST /services.json
   def create
+    if  nil? || volunteer?
```
again, the if ni? is not correct, and these three lines are unneeded.

jrmcgarvey commented on app/controllers/services_controller.rb:
```diff
@@ -13,57 +13,88 @@ class ServicesController < ApplicationController
   # PATCH/PUT /customers/1
   # PATCH/PUT /customers/1.json
   def update
+    if  nil? || volunteer?
```
again, the if nil? is not correct, and these three lines are unneeded.

lena-aiu replied:
Okay, I will make changes

jrmcgarvey commented on app/controllers/services_controller.rb:
```diff
@@ -75,6 +106,15 @@ def update
   # DELETE /customers/1
   # DELETE /customers/1.json
   def destroy
+    if  nil? || volunteer?
```
same deal with the if nil?

jrmcgarvey reviewed (changes requested):
Elena, I apologize, but I did find some more changes that are really needed.  These are marked in the review.  The main problem is the use of if nil?  which is not correct.

lena-aiu commented:
@jrmcgarvey Hi John, I removed an unnecessary part of the code with if nil?, also changed the look of condition for admin and intern. Please review it. Thank you

jrmcgarvey reviewed (approved):
looks ok!
